plugins/http_plugin/beast_http_session.hpp

```diff
@@ -0,0 +1,587 @@
+                    auto session_time = steady_clock::now() - session_begin_;
+                    auto session_time_us = std::chrono::duration_cast<std::chrono::microseconds>(session_time).count();           
+                    fc_dlog(logger, "session time    ${t}", ("t", session_time_us));                            
+                    fc_dlog(logger, "        read    ${t}", ("t", read_time_us_));
+                    fc_dlog(logger, "        handle  ${t}", ("t", handle_time_us_));                                            
+                    fc_dlog(logger, "        write   ${t}", ("t", write_time_us_));                            
+    #endif
+                } catch(...) {
+                    handle_exception();
+                }
+            }    
+
+            void do_read()
+            {
+                read_begin_ = steady_clock::now();
+
+                // Read a request
+                auto self = derived().shared_from_this();
+                http::async_read(
+                    derived().stream(),
+                    buffer_,
+                    *req_parser_,
+                    [self](beast::error_code ec, std::size_t bytes_transferred) { 
+                        self->on_read(ec, bytes_transferred);
+                    }
+                );                
+            }
+
+            void on_read(beast::error_code ec,
+                         std::size_t bytes_transferred)
+            {
+                boost::ignore_unused(bytes_transferred);
+
+                // This means they closed the connection
+                if(ec == http::error::end_of_stream)
+                    return derived().do_eof();
+
+                if(ec) {
+                    return fail(ec, "read");
+                }
+
+                auto req = req_parser_->get();
+
+                handle_begin_ = steady_clock::now();
+                auto dt = handle_begin_ - read_begin_;
+                read_time_us_ += std::chrono::duration_cast<std::chrono::microseconds>(dt).count();
+
+                // Send the response
+                handle_request(std::move(req));
+            }
+
+            void on_write(beast::error_code ec,
+                          std::size_t bytes_transferred, 
+                          bool close)
+            {
+                boost::ignore_unused(bytes_transferred);
+
+                if(ec) {
+                    ec_ = ec;
+                    return fail(ec, "write");
+                }
+
+                auto dt = steady_clock::now() - write_begin_;
+                write_time_us_ += std::chrono::duration_cast<std::chrono::microseconds>(dt).count();
+
+                if(close)
+                {
+                    // This means we should close the connection, usually because
+                    // the response indicated the "Connection: close" semantic.
+                    return derived().do_eof();
+                }
+
+                // create a new parser to clear state 
```

Comment:
I was expecting to see `res_` cleared here as well. Seems like we might hold on to a lot of memory if we do not clear it out.